compair/static/modules/assignment/assignment-module.js

```diff
@@ -178,6 +178,66 @@ module.directive('comparisonPreview', function() {
     };
 });
 
+// module.directive('assignmentStatus', function() {
+// });
+
+module.directive('actionButtons', function() {
+
+    return {
+        scope: false,
+        templateUrl: 'modules/assignment/action-buttons.html',
+        controller: ["$scope", "actionElementPermissions", 
+            function ($scope, actionElementPermissions) {
+                $scope.$watchCollection("[assignment.status, steps_left, comparisons_left, self_evaluation_needed, canManageAssignment]", function(newStatus){
```

Comment:
this should only watch watch's absolutely relevant  like `assignment` and `assignment status`. 

`steps_left`,  `comparisons_left`,  `self_evaluation_needed` are all derived from assignment or assignment status data.

not sure how you want to handle user permissions (`canManageAssignment`). It might be good to pass it along or the condition checker can re-fetch it.